validator/client/src/main/java/tech/pegasys/teku/validator/client/VoluntaryExitDataProvider.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright ConsenSys Software Inc., 2023
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.validator.client;
+
+import tech.pegasys.teku.infrastructure.time.TimeProvider;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.spec.Spec;
+import tech.pegasys.teku.spec.SpecVersion;
+
+public class VoluntaryExitDataProvider {
+  private final Spec spec;
+  private final TimeProvider timeProvider;
+
+  VoluntaryExitDataProvider(final Spec spec, final TimeProvider timeProvider) {
+    this.spec = spec;
+    this.timeProvider = timeProvider;
+  }
+
+  protected UInt64 calculateCurrentEpoch(final UInt64 genesisTime) {
+    final SpecVersion genesisSpec = spec.getGenesisSpec();
```

Comment:
This is very close to the misc helper result. I htink the value in this interface that we were discussing was to have it handling the optional epoch, and doing building out that complexity...
so something to the effect of 
```
SafeFuture<Uint64> getRequiredEpoch(Optional<UInt64> maybeEpochParameter) {
 return maybeEpoch.map(SafeFuture::completedFuture).orElse(() -> 
this::calculateCurrentEpoch // which can use the provider i guess if we have it...
}

SafeFuture<UInt64> calculateCurrentEpoch() {
return genesisDataProvider.getGenesisTime()
.thenApply( genesisTime -> 
final UInt64 slot = miscHelpers.computeSlotAtTime(genesisTime, timeProvider.getCurrentTimeMIllis...)
return spec.computeEpochAtSlot(slot);
}
```

that definitely wont compile... anyway was just illustrative but i thought that was kind of idea was the goal, otherwise after this PR all we've done is implemented a call to the misc helper...

Reply:
My comment was to remove the optional parameter because I don't see the value in handling the optional logic here.

Not a blocker for me either way though.